hazelcast/src/main/java/com/hazelcast/sql/impl/operation/QueryExecuteOperationFragmentMapping.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.operation;
+
+/**
+ * Fragment mapping.
+ */
+public enum QueryExecuteOperationFragmentMapping {
+    /** Fragment members are defined explicitly. */
+    EXPLICIT(1),
+
+    /** Fragment members are data members. */
+    DATA_MEMBERS(2);
+
+    private final int id;
+
+    QueryExecuteOperationFragmentMapping(int id) {
+        this.id = id;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public static QueryExecuteOperationFragmentMapping getById(final int id) {
+        for (QueryExecuteOperationFragmentMapping type : values()) {
```

Comment:
It's better to cache the `values()`, a new array is allocated on every invocation.

Reply:
Fixed.